be/src/runtime/buffer_control_block.cpp

```diff
@@ -257,18 +281,15 @@ void BufferControlBlock::get_batch(GetResultBatchCtx* ctx) {
         return;
     }
     if (!_batch_queue.empty()) {
-        // get result
-        TFetchDataResult* result = _batch_queue.front();
+        std::unique_ptr<SerializeRes> ser = std::move(_batch_queue.front());
         _batch_queue.pop_front();
-        _buffer_rows -= result->result_batch.rows.size();
-        _data_removal.notify_one();
-
-        ctx->on_data(result, _packet_num);
-        _packet_num++;
-
-        delete result;
-        result = nullptr;
-
+        _buffer_bytes -= ser->attachment.length();
+        // _data_removal.notify_one();
+        auto packet_num = _packet_num.load();
+        ++_packet_num;
+        l.unlock();
+        ctx->on_data(ser.get(), packet_num);
+        ser.reset(nullptr);
```

Comment:
`ctx->on_data(ser.release(), packet_num)`, seems we can use this way to simplify code?

Reply:
can't release, if that, we should delete it in on_data(). I will remove L:283.